debezium-core/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java

```diff
@@ -175,4 +181,58 @@ protected void setLogPosition(int index) {
+                "{\"source\":{\"server\":\"my-server\"},\"databaseName\":\"db1\",\"ddl\":\"DROP TABLE foo;\"}");
+        final ProducerRecord<String, String> invalidJSONRecord1 = new ProducerRecord<>(topicName, PARTITION_NO, null,
+                "{\"source\":{\"server\":\"my-server\"},\"position\":{\"filename\":\"my-txn-file.log\",\"position\":39},\"databaseName\":\"db1\",\"ddl\":\"DROP TABLE foo;\"");
+        final ProducerRecord<String, String> invalidJSONRecord2 = new ProducerRecord<>(topicName, PARTITION_NO, null,
+                "\"source\":{\"server\":\"my-server\"},\"position\":{\"filename\":\"my-txn-file.log\",\"position\":39},\"databaseName\":\"db1\",\"ddl\":\"DROP TABLE foo;\"}");
+
+        // Start up the history ...
+        Configuration config = Configuration.create()
+                                            .with(KafkaDatabaseHistory.BOOTSTRAP_SERVERS, kafka.brokerList())
+                                            .with(KafkaDatabaseHistory.TOPIC, topicName)
+                                            .with(DatabaseHistory.NAME, "my-db-history")
+                                            .with(KafkaDatabaseHistory.RECOVERY_POLL_INTERVAL_MS, 1000)
+                                            .with(KafkaDatabaseHistory.consumerConfigPropertyName(
+                                                  ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG),
+                                                  100)
+                                            .with(KafkaDatabaseHistory.consumerConfigPropertyName(
+                                                  ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG),
+                                                  50000)
+                                            .build();
+        history.configure(config, null);
+        history.start();
+
+        final Configuration interuderConfig = Configuration.create()
+                .withDefault(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.brokerList())
+                .withDefault(ProducerConfig.CLIENT_ID_CONFIG, "intruder")
+                .withDefault(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class)
+                .withDefault(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class)
+                .build();
+        try (final KafkaProducer<String, String> producer = new KafkaProducer<>(interuderConfig.asProperties())) {
+            producer.send(nullRecord).get();
+            producer.send(emptyRecord).get();
+            producer.send(noSourceRecord).get();
+            producer.send(noPositionRecord).get();
+            producer.send(invalidJSONRecord1).get();
+            producer.send(invalidJSONRecord2).get();
+        }
+
+        DdlParser recoveryParser = new DdlParserSql2003();
+
+        Tables recoveredTables = new Tables();
+        setLogPosition(1000);
+        history.recover(source, position, recoveredTables, recoveryParser);
```

Comment:
Shouldn't there be some kind of assertion? E.g. making sure a valid message can be consumed after running into some invalid ones? Or am I missing something here?